src/main/java/com/digitalasset/testing/ledger/clock/SandboxTimeProvider.java

```diff
@@ -30,73 +30,60 @@
     return () -> {
       logger.debug("Starting SandboxTimeProvider");
       SandboxTimeProvider p = new SandboxTimeProvider(stub, ledgerId);
-      Monitor monitor = new Monitor();
-      Monitor.Guard started = monitor.newGuard(() -> p.running.get());
-      try {
-        TimeServiceOuterClass.GetTimeRequest req =
-            TimeServiceOuterClass.GetTimeRequest.newBuilder().setLedgerId(ledgerId).build();
-        stub.getTime(
-            req,
-            new StreamObserver<TimeServiceOuterClass.GetTimeResponse>() {
-              public void onNext(TimeServiceOuterClass.GetTimeResponse value) {
-                logger.debug("SandboxTimeProvider received new time {}", value.getCurrentTime());
-                monitor.enter();
-                try {
-                  p.setActualTime(value.getCurrentTime());
-                } finally {
-                  monitor.leave();
-                }
-              }
-
-              public void onError(Throwable t) {
-                logger.warn("SandboxTimeProvider request received an error", t);
-                p.stop(t);
-              }
-
-              public void onCompleted() {
-                logger.warn("SandboxTimeProvider finished");
-                p.stop(null);
-              }
-            });
-
-        monitor.enter();
-        try {
-          boolean running = monitor.waitFor(started, Duration.ofSeconds(30));
-          Preconditions.checkState(running, "SandboxTimeProvider start failed");
-        } finally {
-          monitor.leave();
-        }
-      } catch (InterruptedException ignored) {
-      }
 
+      TimeServiceOuterClass.GetTimeRequest req =
+          TimeServiceOuterClass.GetTimeRequest.newBuilder().setLedgerId(ledgerId).build();
+      stub.getTime(
+          req,
+          new StreamObserver<TimeServiceOuterClass.GetTimeResponse>() {
+            public void onNext(TimeServiceOuterClass.GetTimeResponse value) {
+              logger.debug("SandboxTimeProvider received new time {}", value.getCurrentTime());
+              p.setActualTime(value.getCurrentTime());
+            }
+
+            public void onError(Throwable t) {
+              logger.warn("SandboxTimeProvider request received an error", t);
+              p.stop(t);
+            }
+
+            public void onCompleted() {
+              logger.warn("SandboxTimeProvider finished");
+              p.stop(null);
+            }
+          });
+
+      p.waitForTimeChange(Instant.EPOCH);
+      logger.warn("SandboxTimeProvider started");
```

Comment:
This should be info level
```suggestion
      logger.info("SandboxTimeProvider started");
```